var/spack/repos/builtin/packages/discotec/package.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Discotec(CMakePackage):
+    """This project contains DisCoTec, a code for the distributed sparse
+    grid combination technique with MPI parallelization."""
+
+    homepage = "https://github.com/SGpp/DisCoTec"
+    url = "https://github.com/SGpp/DisCoTec"
+    git = "https://github.com/SGpp/DisCoTec"
+
+    # notify when the package is updated.
+    maintainers = ["freifrauvonbleifrei", "pfluegdk"]
+
+    version("main", branch="main")
+
+    variant("enableft", default=False, description="DisCoTec with algorithm-based fault tolerance")
+    variant("gene", default=False, description="Build for GENE (as task library)")
+    variant("timing", default=True, description="With high-res timers")
+    variant("test", default=True, description="Build Boost tests")
+    variant("openmp", default=False, description="Parallelize with OpenMP")
+    variant("hdf5", default=True, description="Interpolation output with HDF5")
+    variant("vtk", default=False, description="Build with VTK support")
+    variant("lto", default=True, description="Build with link-time optimization")
+    variant(
+        "build_type",
+        default="Release",
+        values=("Release", "RelWithDebInfo", "Debug"),
+        description="Build type for DisCoTec",
+    )
+
+    depends_on("cmake@3.24.2:", type=("build"))
+    depends_on("mpi", type=("build", "run"))
+    depends_on(
+        "boost +test +serialization +filesystem +system +program_options", type=("build", "run")
+    )
+    depends_on("glpk")
+    depends_on("highfive+mpi+boost+ipo", when="+hdf5")
+    depends_on("vtk", when="+vtk")
+
+    def cmake_args(self):
```

Comment:
Use `self.define_from_variant(<option>, <variant>)` below, it's easier.

Reply:
this is a great suggestion, thx!